### PR TITLE
Document stdClass

### LINKED ISSUE #1846: Add page for stdClass built-in class.
As it is referenced in the [predefined](https://www.php.net/manual/en/reserved.classes.php) classes.

That description would also need to be updates as some extension do return stdClasses (e.g. IMAP, JSON) without any typecasting.

### PR DESCRIPTION
**Pull request by kamil-tekiela**
Fixes #1846

### CODE REVIEW

File: language/predefined/versions.xml [hunk 1]
```diff
@@ -159,6 +159,8 @@
 
  <function name="argc" from="PHP 4, PHP 5, PHP 7, PHP 8"/>
  <function name="argv" from="PHP 4, PHP 5, PHP 7, PHP 8"/>
+
+ <function name="stdclass" from="PHP 5, PHP 7, PHP 8"/>
```
Comment (Girgias) [inline language/predefined/versions.xml:163]:
```suggestion
 <function name="stdclass" from="PHP 4, PHP 5, PHP 7, PHP 8"/>
```

**Review by Girgias — COMMENTED** [review]
Other than the nit LGTM, but maybe wait for @cmb69 opinion

File: language/predefined/stdclass.xml [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+   &reftitle.intro;
+   <para>
+    A generic empty class with dynamic properties.
+   </para>
+
+   <para>
+    Objects of this class can be instantiated with
+    <link linkend="language.oop5.basic.new">new</link> operator or created by
+    <link linkend="language.types.object.casting">typecasting to object</link>.
+    Several PHP functions also create instances of this class e.g.,
```
Comment (cmb69) [inline language/predefined/stdclass.xml:20]: I think the comma should be moved before the "e.g.", i.e.
```suggestion
    Several PHP functions also create instances of this class, e.g.
```

File: language/predefined/stdclass.xml [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+    <classsynopsisinfo>
+     <ooclass>
+      <classname>stdClass</classname>
+     </ooclass>
+    </classsynopsisinfo>
+   </classsynopsis>
+
+   <para>
+    This class has no properties or methods.
```
Comment (cmb69) [inline language/predefined/stdclass.xml:56]: Maybe clarify:
```suggestion
    This class has no methods or default properties.
```

File: language/predefined/stdclass.xml [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    Several PHP functions also create instances of this class e.g.,
+    <function>json_decode</function>, <function>mysqli_fetch_object</function>
+    or <methodname>PDOStatement::fetchObject</methodname>.
+   </para>
+
+   <para>
+    Despite not implementing
+    <link linkend="object.get">__get()</link>/<link linkend="object.set">__set()</link>
+    magic methods, this class allows dynamic properties and does not require the
+    <code>#[\AllowDynamicProperties]</code> attribute.
```
Comment (cmb69) [inline language/predefined/stdclass.xml:29]: I think we need to document the builtin attributes, and link to their descriptions, but for now this is okay.

File: language/predefined/versions.xml [hunk 5]
```diff
@@ -159,6 +159,8 @@
 
  <function name="argc" from="PHP 4, PHP 5, PHP 7, PHP 8"/>
  <function name="argv" from="PHP 4, PHP 5, PHP 7, PHP 8"/>
+
+ <function name="stdclass" from="PHP 4, PHP 5, PHP 7, PHP 8"/>
```
Comment (cmb69) [inline language/predefined/versions.xml:163]: I'm not sure whether `stdClass` is available as of PHP 4.0.0, but can't be bothered to check that. So fine for me.

**Review by cmb69 — COMMENTED** [review]
Thank you very much!

**Review by cmb69 — APPROVED** [review]
Thank you!

**Pull request merged**